ngraph/core/reference/include/ngraph/runtime/reference/non_max_suppression.hpp

```diff
@@ -0,0 +1,43 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <functional>
+#include <map>
+#include <ngraph/runtime/host_tensor.hpp>
+#include "ngraph/coordinate_transform.hpp"
+#include "ngraph/op/util/op_types.hpp"
+#include "ngraph/ops.hpp"
+#include "ngraph/shape_util.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            void non_max_suppression(const op::v5::NonMaxSuppression* nms5,
```

Comment:
We never pass operation to reference implementation, I don't want to introduce this new approach.
I propose to avoid it, I think that we need to think how avoid passing of operation to reference implementation. In other case we can use this reference implementation only for NMS 5

Reply:
Now such passing is not performed into reference implementation. But to simplify calculations and to avoid templates, we need some preprocessing of input data of NMS-5, perform calculations for some precision (e.g., for `float`), and perform postprocessing of results of calculations. And to perform preprocessing, we need pass pointer to operation into the preprocessing function.